core/base/integralLines/IntegralLines.h

```diff
@@ -72,9 +173,157 @@ namespace ttk {
       direction_ = direction;
     }
 
-    int preconditionTriangulation(
-      ttk::AbstractTriangulation *triangulation) const {
-      return triangulation->preconditionVertexNeighbors();
+    void findNextVertex(ttk::SimplexId &vnext,
+                        ttk::SimplexId &fnext,
+                        std::vector<ttk::SimplexId> &component,
+                        const SimplexId *offsets) const;
+
+#ifdef TTK_ENABLE_MPI
+
+    /**
+     * @brief Constructs the global identifiers for vertices and edges.
+     *
+     * @tparam triangulationType
+     * @param globalVertexId vector of global identifiers for vertices
+     * @param globalCellId vector of global identifiers for cells
+     * @param seedIdentifiers linked list of identifiers of seed, one for each
+     * integral line
+     * @param forkIdentifiers linked list of identifiers of forks, one for each
+     * integral line
+     * @param trajectories linked list of identifiers of forks, one for each
+     * integral line
+     * @param localVertexIdentifiers linked list of identifiers, one for each
+     * vertex of an integral lines. It represents the number of the vertex in
+     * the integral line.
+     * @param triangulation Triangulation
+     * @return int
+     */
+    template <typename triangulationType>
+    int getGlobalIdentifiers(
```

Comment:
are all these parameters required to be non-const?
please declare in `const` all the parameters which are read-only (that helps the interpretation of the code).
thanks for double-checking all function prototypes in the base code.